src/python/test/test_dxclient.py

```diff
@@ -3022,6 +3023,22 @@ def test_dx_run_workflow(self):
         self.assertIn(analysis_id, new_workflow_desc)
         self.assertIn(stage_id, new_workflow_desc)
 
+        # Setting the input linking to workflowInputSpec
+        dxpy.api.workflow_update(workflow_id,
+                                 {"editVersion": 2,
+                                  "workflowInputSpec": self.wf_input,
+                                  "stages": {stage_id: {'input': {'number': {'$dnanexus_link': {'workflowInputField': 'foo'}}}}}})
+        run("dx describe " + workflow_id)
+        analysis_id = run("dx run " + workflow_id + " -ifoo=474 -y --brief")
+        self.assertTrue(analysis_id.startswith('analysis-'))
+        analysis_desc = run("dx describe " + analysis_id)
+        self.assertIn('foo', analysis_desc)
+        analysis_desc = json.loads(run("dx describe --json " + analysis_id ))
+        self.assertTrue(analysis_desc["runInput"], {"foo": 747})
+        time.sleep(2) # May need to wait for job to be created in the system
+        job_desc = run("dx describe " + analysis_desc["stages"][0]["execution"]["id"])
+        self.assertIn(' number = 474', job_desc)
+
```

Comment:
(not blocking), but if you'd like, may want to add a test to ensure error on including a stage input when it's not there